src/Arcus.Security.Secrets.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -47,13 +56,18 @@ public KeyVaultSecretProvider(IKeyVaultAuthenticator authenticator, IKeyVaultCon
         /// <param name="secretName">The secret name</param>
         /// <returns>The value, stored in Key Vault</returns>
         /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        /// <exception cref="KeyVaultErrorException">The call for a secret resulted in an invalid response</exception>
         public async Task<string> Get(string secretName)
         {
             Guard.NotNullOrEmpty(secretName, nameof(secretName));
             try
             {
-                var keyVaultClient = await GetClientAsync();
-                var secretBundle = await keyVaultClient.GetSecretAsync(VaultUri, secretName);
+                KeyVaultClient keyVaultClient = await GetClientAsync();
+                SecretBundle secretBundle =
+                    await ThrottleTooManyRequests(
+                        () => keyVaultClient.GetSecretAsync(VaultUri, secretName));
+                
+                // TODO: should we allow to return 'null'?
```

Comment:
If Key Vault says that the secret exist it's enough for us, it's not up to use to validate if the secret value is considered valid

Reply:
Ok